JamGame/Assets/Scripts/Overlay/Manager.cs

```diff
@@ -15,23 +16,37 @@ public void ApplyOverlay<O>(O overlay)
     [AddComponentMenu("Scripts/Overlay/Overlay.Manager")]
     public class Manager : MonoBehaviour, IOverlayManager
     {
-        private List<IOverlayRenderer> overlayRenderers;
+        [SerializeField]
+        [ValidateInput(nameof(ValidateOverlays))]
+        private GameObject[] staticOverlayRenderers;
+
+        private IOverlayRenderer[] dynamicOverlayRenderers;
+
+        private IEnumerable<IOverlayRenderer> OverlayRenderers =>
+            dynamicOverlayRenderers.Concat(
+                staticOverlayRenderers.Select(x => x.GetComponent<IOverlayRenderer>())
```

Comment:
Run all the `GetComponent`s once at `Start`

Reply:
Also store resulting `IOverlayRenderer`s in `HashSet` and check that when you fetch `dynamicOverlayRenderers` there's no duplicates